File: app/src/main/java/com/rachelleignacio/listbucket/domain/interactors/impl/DeleteListInteractorImpl.java
package com.rachelleignacio.listbucket.domain.interactors.impl;

import com.rachelleignacio.listbucket.db.DbInteractor;
import com.rachelleignacio.listbucket.domain.executor.Executor;
import com.rachelleignacio.listbucket.domain.executor.MainThread;
import com.rachelleignacio.listbucket.domain.interactors.DeleteListInteractor;
import com.rachelleignacio.listbucket.domain.interactors.base.AbstractInteractor;
import com.rachelleignacio.listbucket.domain.models.List;

/**
 * Created by rachelleignacio on 3/10/17.
 */

public class DeleteListInteractorImpl extends AbstractInteractor implements DeleteListInteractor {
    private DeleteListInteractor.Callback callback;
    private DbInteractor database;
    private List listToDelete;
    private int position;

    public DeleteListInteractorImpl(Executor threadExecutor, MainThread mainThread, Callback callback,
                                    DbInteractor db, List listToDelete, int adapterPosition) {
        super(threadExecutor, mainThread);
        this.callback = callback;
        this.database = db;
        this.listToDelete = listToDelete;
        this.position = adapterPosition;
    }

    @Override
    public void run() {
        database.deleteList(listToDelete);
        mainThread.post(new Runnable() {
            @Override
            public void run() {
                callback.onListDeleted(position);
            }
        });
    }
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/domain/interactors/impl/DeleteListInteractorImpl.kt
package com.rachelleignacio.listbucket.domain.interactors.impl

import com.rachelleignacio.listbucket.db.DbInteractor
import com.rachelleignacio.listbucket.domain.executor.MainThread
import com.rachelleignacio.listbucket.domain.executor.ThreadExecutor
import com.rachelleignacio.listbucket.domain.interactors.DeleteListInteractor
import com.rachelleignacio.listbucket.domain.interactors.base.AbstractInteractor
import com.rachelleignacio.listbucket.domain.models.List

/**
 * Created by rachelleignacio on 8/9/17.
 */
class DeleteListInteractorImpl(threadExecutor: ThreadExecutor,
                               mainThread: MainThread,
                               private val callback: DeleteListInteractor.Callback,
                               private val dbInteractor: DbInteractor,
                               private val listToDelete: List,
                               private val adapterPosition: Int
) : AbstractInteractor(threadExecutor, mainThread), DeleteListInteractor {

    override fun run() {
        dbInteractor.deleteList(listToDelete)
        mainThread.post(Runnable { callback.onListDeleted(adapterPosition) })
    }
}